### PR TITLE
Preserve the UTF8 BOM in localized files

### PR DESCRIPTION
**Pull request by ladipro**
### Problem
Template Localizer removes the UTF8 BOM from localized files. This shows as an extra annoying diff hunk.

### Solution
Made the code preserve the BOM in the target file. We still don't add it to files that don't already have it, consistent with the Unicode Standard recommendation.

### Checks:
- [x] Added unit tests
- [ ] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

**Review by vlada-shubina — APPROVED** [review]
Thanks!

File: src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringUpdater.cs [hunk 1]
```diff
@@ -157,6 +165,28 @@ await SaveTemplateStringsFileAsync(strings, existingStrings, locFilePath, forceU
...
+        /// <returns>A task that represents the asynchronous operation.</returns>
+        private static async Task TruncateFileWhilePreservingBom(FileStream fileStream, CancellationToken cancellationToken)
+        {
+            byte[] preamble = new byte[Utf8Bom.Length];
+            int offset = 0, read = 0;
+            // Read bytes from the stream until we fill the preamble array or hit EOF.
+            do
+            {
+                read = await fileStream.ReadAsync(preamble, offset, preamble.Length - offset, cancellationToken).ConfigureAwait(false);
+                offset += read;
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringUpdater.cs:183]:
```suggestion
                offset += read;
                // Optimization to not call .ReadAsync twice
                if (offset == preamble.Length)
                {
                    break;
                }
```

**Review by DavidKarlas — APPROVED** [review]
Looks good, suggested small optimization, since 99% of time it will be read in one go

File: src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringUpdater.cs (hunk 1 above)
Comment (ladipro) [reply src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringUpdater.cs:183]: Good point! Committed.

**Pull request merged**